scrapy/spiders/__init__.py

```diff
@@ -14,6 +14,20 @@
 from scrapy.utils.deprecate import method_is_overridden
 
 
+def basespider(decorated_cls):
+    """Marks a :class:`~scrapy.spiders.Spider` subclass as a :ref:`base spider
+    <base-spiders>`."""
+
+    @classmethod
+    def is_abstract(cls):
+        if cls is decorated_cls:
+            return True
+        return super(decorated_cls, cls).is_abstract()
+
+    decorated_cls.is_abstract = is_abstract
```

Comment:
Naming nitpick: we talk about base spiders in docs, but there a method is called is_abstract. 
Not sure I like `is_basespider` though.